LiveStream/Models/LiveStreamEvent.swift

```diff
@@ -0,0 +1,154 @@
+    public let token: String
+  }
+
+  public struct User {
+    public let isSubscribed: Bool
+  }
+}
+
+public func == (lhs: LiveStreamEvent, rhs: LiveStreamEvent) -> Bool {
+  return lhs.id == rhs.id
+}
+
+extension LiveStreamEvent: Decodable {
+  static public func decode(_ json: JSON) -> Decoded<LiveStreamEvent> {
+    let create = curry(LiveStreamEvent.init)
+    return create
+      <^> json <| "creator"
+      <*> json <| "firebase"
+      <*> json <| "id"
+      <*> json <|? "opentok"
+      <*> json <| "stream"
+      <*> (json <| "user" <|> .success(User(isSubscribed: false)))
+  }
+}
+
+extension LiveStreamEvent.Stream: Decodable {
+  static public func decode(_ json: JSON) -> Decoded<LiveStreamEvent.Stream> {
+    let create = curry(LiveStreamEvent.Stream.init)
+    let tmp1 = create
+      <^> json <| "background_image_url"
+      <*> json <| "description"
+      <*> json <| "has_replay"
+      <*> json <|? "hls_url"
+    let tmp2 = tmp1
+      <*> json <| "is_rtmp"
+      <*> json <| "is_scale"
+      <*> json <| "live_now"
+      <*> json <| "max_opentok_viewers"
+    let tmp3 = tmp2
+      <*> json <| "name"
+      <*> json <| "project_web_url"
+      <*> json <| "project_name"
+      <*> json <|? "replay_url"
+    return tmp3
+      <*> (json <| "start_date" >>- toDate)
+      <*> json <| "web_url"
+  }
+}
+
+extension LiveStreamEvent.Creator: Decodable {
+  static public func decode(_ json: JSON) -> Decoded<LiveStreamEvent.Creator> {
+    return curry(LiveStreamEvent.Creator.init)
+      <^> json <| "creator_avatar"
+      <*> json <| "creator_name"
+  }
+}
+
+extension LiveStreamEvent.Firebase: Decodable {
+  static public func decode(_ json: JSON) -> Decoded<LiveStreamEvent.Firebase> {
+    let create = curry(LiveStreamEvent.Firebase.init)
+    let tmp = create
+      <^> json <| "firebase_api_key"
+      <*> json <| "chat_path"
+      <*> json <| "green_room_path"
+    return tmp
+      <*> json <| "hls_url_path"
+      <*> json <| "number_people_watching_path"
+      <*> json <| "firebase_project"
+      <*> json <| "scale_number_people_watching_path"
+  }
+}
+
+extension LiveStreamEvent.OpenTok: Decodable {
+  static public func decode(_ json: JSON) -> Decoded<LiveStreamEvent.OpenTok> {
+    return curry(LiveStreamEvent.OpenTok.init)
+      <^> json <| "app"
+      <*> json <| "session"
+      <*> json <| "token"
+  }
+}
+
+extension LiveStreamEvent.User: Decodable {
+  static public func decode(_ json: JSON) -> Decoded<LiveStreamEvent.User> {
+    return curry(LiveStreamEvent.User.init)
+      <^> json <| "is_subscribed"
+  }
+}
+
+private let dateFormatter: DateFormatter = {
+  let dateFormatter = DateFormatter()
+  dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+  return dateFormatter
+}()
+
+private func toDate(dateString: String) -> Decoded<Date> {
+
+  guard let date = dateFormatter.date(from: dateString) else {
+    return .failure(DecodeError.custom("Unable to parse date format"))
```

Comment:
so i can't remember - does this break the decoding if it's a failure? do we want this?